### DISCUSSION TITLE
Playback Speeds

### DISCUSSION THREAD

**Discussion by ThajASara** [PaulWoitaschek/Voice#1316] [Ideas]
So far I am loving this app.  The only issue for me is the highest playback speed is 2x. I mostly listen to things faster than this (2.5-3x typically).  I would love to see this added.

**Comment by dalb8** [comment]
It could also be argued that finer control is needed at the speeds < 1.0x e.g. 0.85x. Perhaps the same UI as Jump To Minute should be used.

**Comment by ThajASara** [comment]
I know one really good implementation I have seen is in the AntennaPod podcatcher, which is also ope source.  They use the Presstissimo program to do it, but you can select the various playback speeds you want in your quick menu so you can choose the ones you use the most often and cycle through them quickly on the player.

**Comment by intrications** [comment]
AntennaPod is discussing switching away from Prestissimo to an internal implementation.

https://github.com/AntennaPod/AntennaPod/issues/532

**Comment by dalb8** [comment]
Hey, I use Browser Intercept every day! Can't believe it's not built into Android; what a great app.

MAP is using Java instead of C for the speed shifting as of the latest version.

-------- Message d'origine --------
De : Michael Basil notifications@github.com
Envoyé: 19 juillet 2015 15:11:26 UTC+01:00
À : Ph1b/MaterialAudiobookPlayer MaterialAudiobookPlayer@noreply.github.com
Cc : dalb8 daithi_black@yahoo.ie
Objet : Re: [MaterialAudiobookPlayer] Playback Speeds (#255)

AntennaPod is discussing switching away from Prestissimo to an internal implementation.

https://github.com/AntennaPod/AntennaPod/issues/532

---

**Comment by PaulWoitaschek** [comment]
What do you mean by browser intercept?

**Comment by intrications** [comment]
@dalb8 Thanks. Glad to hear it is useful.

@Ph1b Browser Intercept is just a little utility app I wrote a couple of years ago:
https://github.com/intrications/browser-intercept
https://play.google.com/store/apps/details?id=com.intrications.android.sharebrowser

**Comment by PaulWoitaschek** [comment]
I changed the way it works, so you can set a more fine tuned speed now. And you set in ton the fly, so no need for clicking ok before. Just set your speed and you see if it sounds good while setting it.

About the speed:
I think the very most of the people will never us a speed above 1.5 times. Except if they think it i s funny for a few seconds. So I want to have the interface simple and don't show values most of the pople don't use. Sorry :-/

**Comment by brandonocasey** [comment]
I created a Fork of Voice here with a max playback speed of 6. If anyone else want to use it feel free. Also feel free to bug me to pull changes from master in the issues if I fall behind, I will try to keep this up to date though. Release will be attatched with an apk in the releases section on Github:

https://github.com/BrandonOCasey/Voice
https://github.com/BrandonOCasey/Voice/releases

**Comment by PaulWoitaschek** [comment]
#833 will enable the possibility to use larger playback speeds without ruining the UX.

**Comment by jtagcat** [comment]
How about:
![image](https://user-images.githubusercontent.com/38327267/82599624-2b302380-9b9c-11ea-8b7b-73e7054e7797.png)
I think only one of the following should be chosen:
 - The number + `+` and `-` buttons (what I would prefer)
 - The number + the slider

The small line under the number indicates, that you can edit by clicking.

![image](https://user-images.githubusercontent.com/38327267/82600160-fc667d00-9b9c-11ea-82dd-9278d133c17e.png)

**Comment by HowardWolosky** [comment]
Is this issue still valid?  Should it be closed now?  You have more fine-grained control with a slider similar to what @jtagcat has above, and it goes up to `2.5x`.

**Comment by JanPeldrimovsky** [comment]
> 
> 
> I changed the way it works, so you can set a more fine tuned speed now. And you set in ton the fly, so no need for clicking ok before. Just set your speed and you see if it sounds good while setting it.
> 
> About the speed:
> I think the very most of the people will never us a speed above 1.5 times. Except if they think it i s funny for a few seconds. So I want to have the interface simple and don't show values most of the pople don't use. Sorry :-/

I mostly listen at least at 3x speed. I understand that most users won't listen at such speeds but only 2.5x speed makes Voice less usable for me.

One solution that comes to my mind is to add max speed to preferences. It would be set to 2.5x by default, but could be increased. 

I'm now sure if this could be done easily, I'm not an Android developer, but it would solve the problem and UX would be great for both standard user and high speed reader.

**Comment by HowardWolosky** [comment]
Well, changing the max speed to `3.0x` (without adding a preference to _set_ what the max speed should be) appears to be as easy as modifying this constant:

https://github.com/PaulWoitaschek/Voice/blob/113d54d0f072dd834e383bc625a48f98b182ad6b/data/src/main/java/de/ph1b/audiobook/data/Book.kt#L62-L65

(as found by @brandonocasey in the fork that he made: https://github.com/brandonocasey/Voice/commit/021c5c2001b499c139756f81f43060cd552db59c#diff-66a2715b7a1b6c4a96062e9d54b9a21c)

It looks like @PaulWoitaschek increased the max speed from `2.0` to `2.5` back in 2016 with e772a4dc0209d6fa04bd608fa46d9140d49f026b, but there was no referenced issue for why he increased it back then.  Maybe he'd consider a PR (like Brandon's) that simply increases what the max speed might be?

**Comment by AnnanFay** [comment]
Just posting to chime in that I agree it should be increased.

I **start** listening to audiobooks at 2x and often increase to 3x and 3.5x. I rarely go up to 4x or 5x.

The main reason I listen to audiobooks at 5x+ is that it seems to improve ability to interpret slower stuff in the ~3-4x speed range. And it requires my complete focus on the audio or the mind will start tuning it out. Listening to the same audio repeatedly also helps with training as you continually improve, at least with that specific narrator.

I would avoid changing the limit to 3.0 unless there is a reason not to make it higher.

**Comment by PaulWoitaschek** [comment]
https://github.com/PaulWoitaschek/Voice/discussions/1334#discussioncomment-3310890

Implemented :)
